src/modules/user/adapter/controller/user-profile.controller.ts

```diff
@@ -0,0 +1,23 @@
+import { Body, Controller, Get, Post, UseGuards } from '@nestjs/common';
+import { UserProfileService } from '../../usecase/service/user-profile.service';
+import { SupabaseGuard } from 'src/modules/auth/guard/auth.guard';
+import { AuthUser } from 'src/modules/auth/decorator/auth-user.decorator';
+import { UserProfileInput } from '../../dto/user-profile.input';
+import { UserProfileOutput } from '../../dto/user-profile.output';
+
+@Controller('/user-profile')
+export class UserProfileController {
+    constructor(private readonly userProfileService: UserProfileService) {}
+
+    @Post()
+    @UseGuards(SupabaseGuard)
+    async upsertUserProfile(@AuthUser() { sub: id }: Record<string,string>, @Body() userProfile: UserProfileInput): Promise<void> {
```

Comment:
@PCalape The `Record<string,string>` types in this file, why not define it separate like `UserProfileInput`?

Reply:
Copy on this @arnellebalane!

Reply:
Done applying changes @arnellebalane